### PR TITLE
Bump cobra to 0.10.1

### PR DESCRIPTION
**Pull request by druvus**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This PR is waiting for an updated swiglpk package at conda-forge then the addition of optlang to conda-forge.

### CODE REVIEW

**Comment by druvus** [comment]
swiglpk is finally updated and ready.

File: recipes/cobra/meta.yaml [hunk 1]
```diff
@@ -7,45 +7,50 @@ package:
...
   fn: cobra-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/c/cobra/cobra-{{ version }}.tar.gz
-  sha256: 104e1f73f78d463849161143e8f2b98fcd7cb5ce8ee2c5b0c704ce3153036e30
+  sha256: ee74a9d585eb3fc40b3e8b6f7516bb80bb178c8413cd071c6c3189d93337cce5
 
 build:
-  number: 1
-  skip: True # [osx]
+  number: 0
+#  skip: True # [osx]
```
Comment (mbargull) [inline recipes/cobra/meta.yaml]: If it should be built under `osx`, just remove this line altogether.

File: recipes/cobra/meta.yaml [hunk 2]
```diff
@@ -7,45 +7,50 @@ package:
...
 
 requirements:
   build:
     - python
     - setuptools
```
Comment (mbargull) [inline recipes/cobra/meta.yaml]: Looks like the actual build requirements are only `python` and `setuptools`. Anything else can be removed.

File: recipes/cobra/meta.yaml [hunk 3]
```diff
@@ -7,45 +7,50 @@ package:
...
-    - cobra.design
     - cobra.flux_analysis
     - cobra.io
     - cobra.manipulation
-    - cobra.solvers
     - cobra.test
-    - cobra.topology
+    - cobra.util
 
   requires:
```
Comment (mbargull) [inline recipes/cobra/meta.yaml]: As we don't actually run the tests, we can remove the `test/requires` section. The import of `cobra.test` will still succeed as in `cobra/test/__init__.py` the `pytest` imports are guarded by a `try ... except`.

**Comment by druvus** [comment]
thanks @mbargull I will fix it once optlang is merged. 

Current status of optlang: conda-forge/staged-recipes#4860

**Comment by mbargull** [comment]
Pretty sure this is finished. But since it still has the `WIP` label, I'll leave it to you, @druvus, to clear that label and merge 😉.

**Comment by druvus** [comment]
thanks @mbargull for the review here and at conda-forge!

**Pull request merged**